hiring/hiring-process/engineering-hiring/scoring-a-take-home-exercise.md

```diff
@@ -1,53 +1,58 @@
+| **Separation of concerns**                                                                                                                                                                                                       |       |
+| Business, presentation, and implementation logic are all mixed together (everything in the Activity/View/Fragment).                                                                                                              | 0     |
+| There is some separation of concerns (ex: separate interface and classes for querying data/making network calls).                                                                                                                | 1     |
+| Candidate has used a standard architecture pattern (MV*/others) or a library (Mobius/MvRx/others). The candidate could also have rolled their own architecture as long as there is clear documentation on their custom solution. | 2     |
+| **Code style**                                                                                                                                                                                                                   |       |
+| Candidate has not followed any particular naming conventions throughout the codebase                                                                                                                                             | 0     |
+| Naming conventions are mostly consistent, with some occasional slips                                                                                                                                                             | 1     |
+| A standard naming convention is followed throughout the codebase                                                                                                                                                                 | 2     |
+| **Code maintainability**                                                                                                                                                                                                         |       |
+| Names are unnecessarily short or meaningless. Ex: `a`, `something`, `doWork`, `runQuery`. An exception to this rule would be something like the iteration variable in a `for` loop.                                              | 0     |
+| Names are mostly descriptive and indicative of the intent. Some names might occasionally be unclear without explanation.                                                                                                         | 1     |
+| Names are mostly descriptive and indicative of the intent. Unclear names have clear comments explaining the purpose of the variable/function and the intent behind it.                                                           | 2     |
```

Comment:
It looks like much of Code style and maintainability point to the same thing. I think code style should include things like code formatting and things related to the linter. Whereas maintainability is about comments, naming conventions, code structure using single responsibility principle. Wdyt?

Reply:
I would not like to include code formatting and linting for this exercise, because:

- Most people are not used to thinking about these things at all, and
- We have automated formatting checks that fix things for us anyway.

I want to be more focused on things which developers themselves have figured out they need to do, like clear naming and consistent conventions.

Does this make sense?

Reply:
That's fair. Let's re-review this later.

Reply:
Sasikanth and Rakshak had the same feedback. Removed the `Code Style` section for now and adjusted the score cut-offs to take this into account.